src/dispatch/plugins/dispatch_slack/incident/interactive.py

```diff
@@ -1849,6 +1851,54 @@ def handle_update_incident_submission_event(
     )
 
 
+@app.shortcut(IncidentShortcutCallbacks.report, middleware=[db_middleware, shortcut_context_middleware])
+def report_incident(
+    ack: Ack,
+    body: dict,
+    client: WebClient,
+    context: BoltContext,
+    db_session: Session,
+    shortcut: dict,
+):
+    ack()
+    initial_description = None
+    if body.get("message"):
+        permalink = (
+            client.chat_getPermalink(
+                channel=context["subject"].channel_id, message_ts=body["message"]["ts"]
+            )
+        )["permalink"]
+        initial_description = f"{body['message']['text']}\n\n{permalink}"
+
+    blocks = [
```

Comment:
Everything here and below is the same as in `handle_report_incident_command`, recommend moving into it's own function `draw_report_incident_modal` or similar to keep things consistent.

Reply:
@wssheldon two differences are the `initial_value` for the `description_input` (line 1884) and the `trigger_id` (line 1901).